### DISCUSSION TITLE
Dynamically generate static XML files with nextjs

### DISCUSSION THREAD

**Discussion by jmartinhoj** [vercel/next.js#43214] [Help]
I have a Next.js podcasts website with many user created podcasts, with dynamic routes in `pages/podcast/[slug]/index.js`, which is a page with the episodes of the given podcast. For example, `<WEBSITEURL>/podcast/evening-show` returns a webpage with all the episodes of the Evening Show podcast.

I want every podcast to have its own RSS feed, in order to be able to index to streaming services, like spotify.

Until now, my solution has been having a page in `pages/podcast/[slug]/feed.js` that uses `getInitialProps` to send an XML page to the response object:

```javascript
export default class Rss extends React.Component {
  static async getInitialProps({ res, query }) {
    if (!res) {
      return
    }
    const podcastContent = await getPodcastContent(`/programa/${query.slug}`) //this fetches an object with the podcast's content 
    res.setHeader('Content-Type', 'text/xml')
    res.write(getRssXml(podcastContent)) //this returns an XML that is written in the response object
    res.end()
  }
}
```

However, my goal is to use Incremental Static Generation in order to generate these XML files, so that my website doesn't have to fetch the backend API everytime it receives a request for the feeds.

Many online solutions include the RSS generation on `getStaticProps` and write it to an XML file using `fs.WriteFile()` to write to the `public` folder. However, this solution doesn't work for user created content like podcasts, because all the `public` content must be accessible during build time.

**Comment by allanlasser** [comment]
I see you're using the `pages` directory in your example, and not sure the result you're looking for is doable within that system.

This is achievable, however, with [Route Handlers](https://beta.nextjs.org/docs/routing/route-handlers) in Next 13.2. You can generate XML files from a username or ID in the route slug, then have the results cached as static data.

I've written up how I added XML feeds to my Next 13.2 site by using route handlers: https://allanlasser.com/posts/generating-feeds-with-next-js-route-handlers

Hope this helps!

**Reply by JamesIves** [reply]
Thank you for writing this! Was having a really hard time finding an example of this using the apps directory.
